be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -304,138 +347,234 @@ Status JsonReader::close() {
  *      value2     30
  */
 Status JsonReader::read_chunk(Chunk* chunk, int32_t rows_to_read, const std::vector<SlotDescriptor*>& slot_descs) {
-    // Only operator != is impletement for simdjson::ondemand::document_stream::iterator.
-    if (!(_doc_stream_itr != _doc_stream.end())) {
-        RETURN_IF_ERROR(_read_and_parse_json());
+    simdjson::ondemand::object row;
+    auto st = Status::OK();
+
+    std::vector<SlotDescriptor*> reordered_slot_descs(slot_descs);
+    for (int32_t n = 0; n < rows_to_read; n++) {
+        st = _next_row();
+        if (!st.ok()) {
+            if (st.is_end_of_file()) {
+                return st;
+            }
+            chunk->set_num_rows(n);
+            _counter->num_rows_filtered++;
+            _state->append_error_msg_to_file("", st.to_string());
+            return st;
+        }
+
+        st = _get_row(&row);
+        if (!st.ok()) {
+            chunk->set_num_rows(n);
+            _counter->num_rows_filtered++;
+            _state->append_error_msg_to_file("", st.to_string());
+            return st;
+        }
+
+        if (n == 0 && _scanner->_json_paths.empty() && _scanner->_root_paths.empty()) {
+            // Try to reorder the column according to the column order of first json row.
+            // It is much faster when we access the json field as the json key order.
+            _reorder_column(&reordered_slot_descs, row);
+            row.reset();
+        }
+
+        st = _construct_row(&row, chunk, reordered_slot_descs);
+        if (!st.ok()) {
+            chunk->set_num_rows(n);
+            _counter->num_rows_filtered++;
+            _state->append_error_msg_to_file("", st.to_string());
```

Comment:
Do not append too many error messages to it, otherwise, it's too hard for the user to check it and may cost many disk spaces.

Reply:
Updated.